jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/HTTP3Stream.java

```diff
@@ -372,6 +372,8 @@ public void onTrailer(HeadersFrame frame)
 
     public void onFailure(long error, Throwable failure)
     {
+        HTTP3StreamConnection connection = (HTTP3StreamConnection)endPoint.getConnection();
+        connection.onFailure(error, failure);
```

Comment:
Don't think it's the right place for this code.
Should all be in H3StreamConnection.

Reply:
The problem is that `HTTP3StreamConnection` doesn't get notified in any way. But `HttpStreamOverHTTP3.onFailure()` does get called and looks like the right place to forward the failure to `HTTP3StreamConnection`.

Let me modify this.